src/angular/loader/loader.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable, Type, ComponentRef } from '@angular/core';
+import { LoaderComponent } from './loader.component';
+
+@Injectable()
+export class LoaderService {
+
+    constructor() {}
+
+    public registeredLoaders = {};
+
+    public register(name: string, loader: LoaderComponent) {
+        if (!this.registeredLoaders[name]) {
+            this.registeredLoaders[name] = loader;
+        }
+    }
+
+    public unregister(name: string) {
+        if (this.registeredLoaders[name]) {
+            delete this.registeredLoaders[name];
+        }
+    }
+
+    public activate(name: string = 'general') {
```

Comment:
Maybe add an attribute to the service called `mainLoaderName` that will hold the name of the main (default) registered loader name (`general`). And this function will have the name parameter default to that attribute value.